### PR TITLE
Action task supports multi GPU training

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary
This PR enables multi GPU training to action task.
Current multi GPU training needs to be refactored because code flow for multi GPU enablement in each tasks are all different.
I'll refactored it after this pr is merged and at that time I'll implement unit test for that.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [x] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by jaegukhyun — CHANGES_REQUESTED** [review]
LGTM!. Could you add multi gpu test for action detection?

**Comment by eunwoosh** [comment]
> LGTM!. Could you add multi gpu test for action detection?

Thanks! it's not hard but my intention is adding as little as possible test to integration test to reduce test time.
Is it necessary to add test in detection task also? If yes, then I'll add it too.

**Comment by jaegukhyun** [comment]
> > LGTM!. Could you add multi gpu test for action detection?
> 
> Thanks! it's not hard but my intention is adding as little as possible test to integration test to reduce test time. Is it necessary to add test in detection task also? If yes, then I'll add it too.

It make sense to me. Could you add multi-gpu test for action classification and action detection in e2e tests?
I think at least e2e should check all TC

**Comment by eunwoosh** [comment]
> It make sense to me. Could you add multi-gpu test for action classification and action detection in e2e tests? I think at least e2e should check all TC

I added it.

**Review by jaegukhyun — DISMISSED** [review]

File: otx/algorithms/action/adapters/mmaction/task.py [hunk 1]
```diff
@@ -196,6 +197,40 @@ def configure(
         self._config = recipe_cfg
         return recipe_cfg
 
+    def _configure_device(self, cfg, training):
```
Comment (sungmanc) [inline otx/algorithms/action/adapters/mmaction/task.py]: Could you add the proper type-hint for this method?

Comment (eunwoosh) [reply otx/algorithms/action/adapters/mmaction/task.py]: I updated it :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ All notable changes to this project will be documented in this file.
 
 ### New features
 
--
+- Action task supports multi GPU training.
```
Comment (goodsong81) [inline CHANGELOG.md]: Please add PR link :)

**Review by goodsong81 — CHANGES_REQUESTED** [review]
LGTM, but could you please fix one thing below?

File: CHANGELOG.md (hunk 2 above)
Comment (eunwoosh) [reply CHANGELOG.md]: Thanks for pointing it. I added it.

**Review by jaegukhyun — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (eunwoosh) [reply CHANGELOG.md]: Could you approve my PR if it's ok?

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**